lib/open_food_network/option_value_namer.rb

```diff
@@ -72,5 +71,39 @@ def scale_for_unit_value
 
       unit
     end
+
+    def pluralize(unit_name, count)
+      I18nUnitNames.instance.pluralize(unit_name, count)
+    end
+
+    # Provides efficient access to unit name inflections.
+    # The singleton property ensures that the init code is run once only.
+    # The OptionValueNamer is instantiated in loops.
+    class I18nUnitNames
+      include Singleton
+
+      def pluralize(unit_name, count)
+        return unit_name if count.nil?
+
+        @unit_keys ||= unit_key_lookup
+        key = @unit_keys[unit_name.downcase]
+
+        return unit_name unless key
+
+        I18n.t(key, scope: "unit_names", count: count, default: unit_name)
+      end
+
+      private
+
+      def unit_key_lookup
```

Comment:
I think this is overcomplicated. Why not just load the key/value we need?

Reply:
Yeah, I need to add some documentation and explain it step by step. So the user enters "fardo". We can't look up "fardo" with I18n because there is no key named that. So we need to look up which key relates to "fardo" and we find `:bale`. Then we can translation `t(:bale, count: 2)` and get "fardos".

Or did you understand this part and you are hinting at something else?

Reply:
ok, I didnt understand that. It's not straight forward. Now I understand why this is necessary :+1: